### PR TITLE
Port to Qt6

### PR DESCRIPTION
**Pull request by lkorbel**
This version works with Qt6.
Also few more data types added.

### CODE REVIEW

**Comment by lkorbel** [comment]
@sierdzio Hi Tom! Sending my current modifications to the module. Do you plan to continue public releases for this one?

**Comment by sierdzio** [comment]
> @sierdzio Hi Tom! Sending my current modifications to the module. Do you plan to continue public releases for this one?

Yes, when we will do any changes, they will be pushed here also. But since we're mostly busy with external projects now, nobody is paying much attention to MCDB stuff :-(

File: CMakeLists.txt [hunk 1]
```diff
@@ -42,21 +45,20 @@ target_compile_definitions(mconfig PUBLIC MCONFIG_LIB)
 
 option(mconfig-crypto "MConfig: Build with encryption functions" OFF)
 if(mconfig-crypto)
+  message("MConfig: using encryption library")
+  target_compile_definitions(mconfig PRIVATE CRYPTED_CONFIG)
```
Comment (sierdzio) [inline CMakeLists.txt]: @lkorbel should probably be `ENCRYPTED_CONFIG`.

Also, similar changes will be required for qmake support (.pri file)

Comment (lkorbel) [reply CMakeLists.txt]: @sierdzio Thank you, I will rename it.

**Comment by sierdzio** [comment]
OKi let's merge. I guess we'll just abandon our GitLab copy and move development here, if it gets restarted

**Pull request merged**